src/main/java/com/google/sps/servlets/DeleteMarker.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that process information about user's markers. */
+@SuppressWarnings("serial")
+@WebServlet("/delete_marker")
+public class DeleteMarker extends HttpServlet {
+  /** Processes POST request by deleting marker by received id. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get Id from the request.
+    try {
+      String contentId = request.getParameter("contentId");
+      Long id = Long.parseLong(contentId);
+
+      Key markerKey = KeyFactory.createKey("Marker", id);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.delete(markerKey);
```

Comment:
Nothing to do for this PR, except perhaps to adding a TODO. At some point you will have to store the user ID associated with the marker to know which markers were stored by which users. You should remember to check that the marker belongs to the currently logged in user before deleting it to stop users from deleting other people's markers by guessing the ID.

Reply:
I added the user to the Route entity properties and I think the delete function will no longer be necessary.

Reply:
Actually i thought about that, and if we wouldn't delete markers from datastore (when corresponding route deleted or user delete that marker from the route) it will lead to enourmous size of used memory.

Reply:
There are multiple options. One is to keep the existing individual delete, and check whether the right person is deleting. Another option is to only delete the markers locally, and when a user saves a route, making all the changes in a batch on the server, including deleting markers that no longer exist.